Sources/PackageGraph/Pubgrub/PubgrubDependencyResolver.swift

```diff
@@ -838,7 +843,7 @@ private final class DiagnosticReportBuilder {
             if let derivedLine = derivedLine {
                 write(
                     incompatibility,
-                    message: "because \(description(for: ext)) and \(description(for: derived)) (\(derivedLine)), \(incompatibilityDesc).",
+                    message: "Because \(description(for: ext)) and \(description(for: derived)) (\(derivedLine)), \(incompatibilityDesc).",
```

Comment:
I wonder if it would be more clear to lead with the incompatibility rather than putting it at the very end, but the new wording in the composed string is definitely an improvement, and I don't know that turning the order around would work for all cases.  But it might be something to consider for a future proposal.

In an IDE, each of these clauses should ideally be clickable where it takes you to the range of text in the package manifest that declares each dependency.  But we're far from having enough information to show that yet.

Reply:
> In an IDE, each of these clauses should ideally be clickable where it takes you to the range of text in the package manifest that declares each dependency. But we're far from having enough information to show that yet.

totally agree, unfortunately it does not look like the infrastructure to point the source location is in place to support this 

> I wonder if it would be more clear to lead with the incompatibility rather than putting it at the very end, but the new wording in the composed string is definitely an improvement, and I don't know that turning the order around would work for all cases. But it might be something to consider for a future proposal.

Yes, I also feel the same way. I will try to see if I can tweak this further

Reply:
Right, the infrastructure for capturing locations is definitely not there yet.  One approach I've wanted to experiment with would be to add an implicit `#line` argument to each of the declarations in the PackageDescription module so that we would at least get line information, and then pass that back through the JSON to SwiftPM.  I haven't yet found time to try that out but I think that would give us at least rough locations in the manifest for each of the creations of the dependencies.  This could also be used for creation of targets, products, etc, and other objects in the manifest.  It might also be possible to build the manifest for debugging and then get callsites that way, but it doesn't seem trivial either way.

Reply:
> I wonder if it would be more clear to lead with the incompatibility rather than putting it at the very end, but the new wording in the composed string is definitely an improvement, and I don't know that turning the order around would work for all cases. But it might be something to consider for a future proposal.

@abertelrud I took another (and more aggressive) swing at it, now reversing the order of how the "argument" is built. changes to test cases verification text probably are best indication of the outcome

Reply:
That's great — I think this is a big improvement.  The messages are still pretty gnarly in many cases, but I think it looks a lot better.

Reply:
>  The messages are still pretty gnarly in many cases

I agree, but that would require much deeper refactoring of the resolver engine

Reply:
Absolutely, that's for another PR.